submissions/morenko/memory-game/js/index.js

```diff
@@ -0,0 +1,182 @@
+    createCards(); // заполняем дом массивом карт
+    countCards = cards.length;
+}
+
+// // функция перезагрузки карт в дом, предварительно перемешиваем их
+const reloadBoard = function() {
+    board.innerHTML = ''; // очищаем дом
+    loadCards(); // загружаем карты
+}
+
+// функция переворота карты
+// чтобы перевернуть карту при клике на нее, нужно добавить класс flip на карточку (div)
+let firstClick = null;
+function flipCard({target}) {
+    // на входе ждем елемент на котором произошло событие клик (назовем его target)
+    // по умолчанию на всех картах есть класс closed, и при первом клике мы будем его удалять
+    const cardClicked = target.closest('.card.closed');
+    // 
+    if (!cardClicked) {
+        // если в карте нет класса closed то выходим, инчего не делаем
+        return;
+    }
+    // если каласс closed есть, то удаляем его и добавляем класс flip
+    cardClicked.classList.remove('closed');
+    cardClicked.classList.add('flip');
+
+    // кликнутая карта становится первой , называем ее firstClick, по умолчанию значение null
+
+    // сделаем проверку, 
+    if (!firstClick) {
+        // если клик по карте первый, то выполняем переворот
+        firstClick = cardClicked;
+    } else {
+        // если не первый, запускаем функцию проверки на инентичность и блокируем доску,
+        // добавляем ей класс blocked
+        checkIdentity(cardClicked);
+        board.classList.add('blocked');
+    }
+}
+
+const checkTimeout = 1000;
+// функция проверки на инентичность
+function checkIdentity(secondClick) {
+    // для проверки будем использовать дата атрибуты (по названию)
+    // создадим константу равенства где сравниваем дата атрибуты первой и второй карты
+    const isEqual = secondClick.dataset.name === firstClick.dataset.name;
+    setTimeout(() => {
+        board.classList.remove('blocked');
+    }, checkTimeout)
+  
+    if (!isEqual) {
+        // если равенство не верно то вызываем функцию, которая возвращает текущей карте исходное положение
+        // и передаем ей карту
+        setTimeout(() => {
+            getStartingPosition(firstClick);
+            getStartingPosition(secondClick);
+            firstClick = null;
+        }, checkTimeout);
+    } else {
+        // если же равенство равно, то оставляем карты перевернутыми, добавляем им клас оупенид
+        secondClick.classList.add('opened');
+        firstClick.classList.add('opened');
+        firstClick = null;
+        // в этом случае отнимаем у счетчика 2
+        countCards -= 2;
+        getWin(); //запускаем поздравление, запустится тогда когда счетчик будет равен 0
+    }
+}
+
+// функция возвращающая текущей карте исходное положение
+function getStartingPosition ( card ) {
+    // на входе ожидает текущую карту
+    card.classList.remove('flip'); // удаляем класс флип
+    card.classList.add('closed'); // добавляем класс клосед
+}
+
+const winTimeout = 7000;
+// функция вывода на экран поздравления
+function getWin() {
```

Comment:
```suggestion
function showWinnerScreen() {
```
`getWin` is meaningless in english :)